WordPress/gutenberg
# Style revisions for Experimental Navigation Component

Fixes #26336

The design for the navigation component has been updated to require a few minor CSS updates; including:

- The individual items should have a height of 32px
- Consecutive groups should have a 24px spacing
- Cancelling native margin styles present in wp-admin environment


## Description
Only some minor CSS additions in `navigation-styles.scss` that only impact the experimental navigation component. Also added new Story to better illustrate changes.

## How has this been tested?
- Viewing the usage of the navigation component within the wc-admin environment
- Viewing the component within Storybook (added new `Groups` Story)
- Running linting/tests

## Screenshots 

Before:

![Screenshot from 2020-10-20 14-44-09](https://user-images.githubusercontent.com/444632/96647581-c90e5280-12e2-11eb-98fc-dcda67b5b3d7.png)

After:

![after](https://user-images.githubusercontent.com/444632/96647600-cf043380-12e2-11eb-8521-5de6d5847e63.png)


## Types of changes
Bug fix, updating to align with newer designs for experimental navigation components


## Test Instructions:
- `npm run storybook:dev`
- Go to the Navigation component.
- Look for spacing between groups in `Groups` story, and shorter items
- To further test the changes in `wc-admin`, use `npm link` to bring in updated `@wordpress/components`
- Observe changes

## Checklist:
